apps/src/storage/dataBrowser/dataUtils.js

```diff
@@ -74,17 +75,45 @@ export function toBoolean(val) {
 
 /**
  * Convert a string to a boolean or number if possible.
- * @param val
+ * @param inputString
  * @returns {string|number|boolean}
  */
-export function castValue(val) {
-  try {
-    return JSON.parse(val);
-  } catch (e) {
-    if (e instanceof SyntaxError) {
-      return val;
+export function castValue(inputString, allowUnquotedStrings) {
+  if (experiments.isEnabled(experiments.APPLAB_DATASETS)) {
+    // 1. Remove leading and trailing whitespace
+    inputString = inputString.trim();
+    // 2. Check for '', undefined, and null
+    if (inputString === '') {
```

Comment:
If I input the value "", will this coerce it to undefined?

Reply:
No, if you type `""`, the argument here will be `'""'`, which will be parsed correctly as `""`

Reply:
The argument here being `'"` only happens when the text input is blank, which should be parsed as `undefined`

Reply:
👍